erts/emulator/beam/index.c

```diff
@@ -58,7 +58,7 @@ IndexTable*
 erts_index_init(ErtsAlcType_t type, IndexTable* t, char* name,
 		int size, int limit, HashFunctions fun)
 {
-    Uint base_size = ((limit+INDEX_PAGE_SIZE-1)/INDEX_PAGE_SIZE)*sizeof(IndexSlot*);
+    Uint base_size = ((limit-1)/INDEX_PAGE_SIZE+1)*sizeof(IndexSlot*);
```

Comment:
This does eliminate the integer overflow problem, but it will also for some values of `limit` unnecessarily round up one extra page. I suggest the following fix instead:
 
` Uint base_size = (((Uint)limit+INDEX_PAGE_SIZE-1)/INDEX_PAGE_SIZE)*sizeof(IndexSlot*);`